src/stdlib/io/file.rs

```diff
@@ -0,0 +1,323 @@
+use crate::{Context, Error, IntoValue, Value};
+use std::{
+    cell::RefCell,
+    fs::File,
+    io::{self, Cursor, Read, Seek, Write},
+    rc::Rc,
+};
+
+use super::std_file_kind::StdFileKind;
+use either::Either;
+use gc_arena::Collect;
+
+fn read_from_any<'gc, W: Write + Read + Seek>(
+    ctx: Context<'gc>,
+    file: &mut W,
+    format: &str,
+    position: &mut usize,
+) -> Result<Option<Value<'gc>>, Error<'gc>> {
+    fn seek_read<'gc, W: Write + Read + Seek>(
+        file: &mut W,
+        position: &mut usize,
+        buf: &mut [u8],
+    ) -> io::Result<usize> {
+        file.seek(io::SeekFrom::Start(*position as u64))?;
+        match file.read(buf) {
+            Ok(n) => {
+                *position += n;
+                Ok(n)
+            }
+            Err(err) => Err(err),
+        }
+    }
```

Comment:
The offset into the file shouldn't need to be manually tracked -- the files track those internally, and Rust already exposes the calls to change the offset through the Seek trait.